src/unix/mod.rs

```diff
@@ -135,7 +135,7 @@ cfg_if! {
     if #[cfg(not(stdbuild))] {
         // cargo build, don't pull in anything extra as the libstd  dep
         // already pulls in all libs.
-    } else if #[cfg(target_env = "musl")] {
+    } else if #[cfg(all(target_env = "musl", target_arch = "x86_64"))] {
```

Comment:
With this cfg, x86-64-musl links statically to libc, but mips-musl links dynamically to libc.